mods/persistence/modules/recycling/recycler.dm

```diff
@@ -0,0 +1,383 @@
+/obj/item/stock_parts/circuitboard/recycler
+	name = "circuitboard (recycler)"
+	build_path = /obj/machinery/recycler
+	board_type = "machine"
+	origin_tech = "{'materials':2}"
+	req_components = list(
+		/obj/item/stock_parts/manipulator = 4,
+		/obj/item/stock_parts/matter_bin = 4,
+		/obj/item/stock_parts/scanning_module = 1)
+	additional_spawn_components = list(
+		/obj/item/stock_parts/keyboard = 1,
+		/obj/item/stock_parts/power/apc/buildable = 1,
+		/obj/item/stock_parts/radio/transmitter/basic/buildable,
+		/obj/item/stock_parts/radio/receiver/buildable,
+	)
+
+///////////////////////
+// Recycler
+///////////////////////
+/obj/machinery/recycler
+	name = "recycler"
+	desc = "Process trash into raw materials."
+	icon = 'icons/obj/recycling.dmi'
+	icon_state = "grinder-o0"
+	layer = ABOVE_HUMAN_LAYER
+	density = TRUE
+	anchored = TRUE
+	use_power = POWER_USE_IDLE
+	idle_power_usage = 1 KILOWATTS		//This is vastly lower than it should for gameplay reasons. A 200hp engine running at 25% power would need 37 kW
+	active_power_usage = 40 KILOWATTS	//This is vastly lower than it should for gameplay reasons. 200 HP electric motor at 100% efficiency would need 149 kW
+	required_interaction_dexterity = DEXTERITY_SIMPLE_MACHINES
+
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = list(
+		/obj/item/stock_parts/power/apc/buildable,
+		/obj/item/stock_parts/radio/transmitter/basic/buildable,
+		/obj/item/stock_parts/radio/receiver/buildable,
+	)
+	stock_part_presets = list(
+		/decl/stock_part_preset/radio/basic_transmitter/recycler = 1, 
+		/decl/stock_part_preset/radio/receiver/recycler = 1,
+	)
+	public_variables = list(
+		/decl/public_access/public_variable/identifier,
+		/decl/public_access/public_variable/use_power,
+		/decl/public_access/public_variable/reagents,
+		/decl/public_access/public_variable/reagents/volumes,
+		/decl/public_access/public_variable/reagents/free_space,
+		/decl/public_access/public_variable/reagents/total_volume,
+		/decl/public_access/public_variable/reagents/maximum_volume,
+	)
+	public_methods = list(
+		/decl/public_access/public_method/toggle_power,
+	)
+
+	var/bloodied 				= FALSE		//Whether we show the sprite with blood or not
+	var/tmp/grinding			= FALSE 	//Whether we're currently busy grinding something down
+	var/tmp/safety_mode 		= 0 		// Temporality stops the machine if it detects a mob
+	var/tmp/efficiency 			= 0.5		//Percentage of materials recovered from the recycled item
+	var/tmp/timelastoutput 		= 0			//Time when the last sheet was outputed from the machine
+	var/const/sound_idle 		= 'sound/machines/creaky_loop.ogg'
+	var/const/sound_processing 	= 'sound/machines/machine_wirr.ogg'
+	var/const/sound_outputs 	= 'sound/machines/hiss.ogg'
+	var/const/sound_reject 		= 'sound/machines/buzz-sigh.ogg'
+	var/tmp/soundid 			= "Recyclers"
+	var/tmp/datum/sound_token/sound_looping
+	var/eat_dir = WEST						//Fixed directio from where the machine will accept trash
```

Comment:
Direction

Reply:
Direction what?

Reply:
"Fixed directio" comment, that is. Should have been more clear, sorry.

Reply:
Oh! Gotcha!